homeassistant/config_entries.py

```diff
@@ -246,6 +246,10 @@ def __init__(
 
         Returns if unload is possible and was successful.
         """
+        if self.source == SOURCE_IGNORE:
+            self.state = ENTRY_STATE_NOT_LOADED
```

Comment:
No need to set this I think ? As we can never change state.

Reply:
Hmm - i'll have to defer to you there, i just did that because all the other success exit paths were doing that. Shall i remove it?